target/linux/rtl838x/dts/rtl8382_allnet_all-sg8208m.dts

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/dts-v1/;
+
+#include "rtl838x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "allnet,all-sg8208m", "realtek,rtl838x-soc";
+	model = "ALLNET ALL-SG8208M";
+
+	aliases {
+		led-boot = &led_sys;
+		led-failsafe = &led_sys;
+		led-running = &led_sys;
+		led-upgrade = &led_sys;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200 mem=128M";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+		reset {
+			label = "reset";
+			gpios = <&gpio0 67 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_sys: sys {
+			label = "sg8208m:green:sys";
+			gpios = <&gpio0 47 GPIO_ACTIVE_HIGH>;
+		};
+		// GPIO 25: power on/off all port leds
+	};
+};
+
+&gpio0 {
+	indirect-access-bus-id = <0>;
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x80000>;
+				read-only;
+			};
+
+			partition@80000 {
+				label = "u-boot-env";
+				reg = <0x80000 0x10000>;
+				read-only;
+			};
+
+			partition@90000 {
+				label = "u-boot-env2";
+				reg = <0x90000 0x10000>;
+				read-only;
+			};
+
+			partition@a0000 {
+				label = "jffs";
+				reg = <0xa0000 0x100000>;
+			};
+
+			partition@1a0000 {
+				label = "jffs2";
+				reg = <0x160000 0x100000>;
+			};
+
+			partition@b2a0000 {
+				label = "firmware";
+				reg = <0x2a0000 0x700000>;
```

Comment:
what about 0x9a0000 till the end?

Reply:
Isnt 0x1000000-2a0000 = d60000? So it could be 0xd60000.

Reply:
well, if there really is nothing else to be protected there, you just use
`<0x2a0000 0xd60000>`

I wonder however what has been the reason for the 0x700000? What vendor partition would be in that area?

Reply:
There was no partition on the device after that. It was probably empty. Maybe they used to have smaller flash chips in the past.

Reply:
Strange, as 0x9a0000 is not really close to a specific flash size.

I'd suggest to use one of the following options:
1. Use "full" flash size and add a comment about the smaller vendor partition to the _commit message_.
2. Use the smaller partition and add a comment afterwards _in the DTS_ telling about the unused space.

I have a weak preference for option 2, as I prefer to not go beyond what the vendor implied.